File: app/src/main/java/com/msd/notebook/common/ProgressBarClass.java
package com.msd.notebook.common;

import android.app.ProgressDialog;
import android.content.Context;

import com.msd.notebook.R;

public class ProgressBarClass {
    private static final ProgressBarClass ourInstance = new ProgressBarClass();
    public Context context;
    private ProgressDialog progressDialog;

    public ProgressBarClass() {
    }
    public static ProgressBarClass getInstance() {
        return ourInstance;
    }

    public void showProgress(Context context) {
        try {
            if (progressDialog != null && progressDialog.isShowing()) {
                return;
            }
            progressDialog = new ProgressDialog(context);
            progressDialog.setMessage(context.getString(R.string.please_wait));
            progressDialog.setProgress(context.getResources().getColor(R.color.indigo_blue));
            progressDialog.setCancelable(false);
            progressDialog.show();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void dismissProgress() {
        try {
            if (null != progressDialog && progressDialog.isShowing())
                progressDialog.dismiss();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

}

File: app/src/main/java/com/msd/notebook/common/ProgressBarClass.kt
package com.msd.notebook.common

import android.app.ProgressDialog
import android.content.Context
import com.msd.notebook.R

class ProgressBarClass {
    var context: Context? = null
    private var progressDialog: ProgressDialog? = null

    fun showProgress(context: Context) {
        try {
            if (progressDialog != null && progressDialog!!.isShowing) {
                return
            }
            progressDialog = ProgressDialog(context)
            progressDialog!!.setMessage(context.getString(R.string.please_wait))
            progressDialog!!.progress = context.resources.getColor(R.color.indigo_blue)
            progressDialog!!.setCancelable(false)
            progressDialog!!.show()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun showProgress(context: Context, message: String) {
        try {
            if (progressDialog != null && progressDialog!!.isShowing) {
                return
            }
            progressDialog = ProgressDialog(context)
            progressDialog!!.setMessage(message)
            progressDialog!!.progress = context.resources.getColor(R.color.indigo_blue)
            progressDialog!!.setCancelable(false)
            progressDialog!!.show()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun dismissProgress() {
        try {
            if (null != progressDialog && progressDialog!!.isShowing) progressDialog!!.dismiss()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    companion object {
        val instance = ProgressBarClass()
    }
}